django_cloud_tasks/models.py

```diff
@@ -0,0 +1,133 @@
+# pylint: disable=no-member
+from datetime import datetime
+from typing import Optional, Dict, List
+from django.db import transaction, models
+from django.db.models import QuerySet, Q
+from django.apps import apps
+from django_cloud_tasks import tasks, serializers
+
+
+class Pipeline(models.Model):
+    name = models.CharField(max_length=100)
+
+    def start(self):
+        routines = self.routines.filter(Q(dependent_routines__id__isnull=True) & ~Q(status=Routine.Statuses.COMPLETED))
```

Comment:
It should be `& Q(status=Routine.Statuses.PENDING))`, because it's the only state possible before starting.

Reply:
I can make a POST request to add new items into this pipeline. I **must** ignore completed leafs